atomicapp/nulecule/container.py

```diff
@@ -31,10 +31,14 @@ def __init__(self, dryrun=False, docker_cli='/usr/bin/docker'):
             except subprocess.CalledProcessError as e:
                 if "client and server don't have same version" in e.output \
                         or "client is newer than server" in e.output:
-                    print("\nThe docker version in this Atomic App differs "
-                          "greatly from the host version.\nPlease use a different "
-                          "Atomic App version for this host.\n")
-                raise e
+                    raise DockerException("\nThe docker version in this Atomic App "
+                                          "differs greatly from the host version.\n"
+                                          "Please use a different Atomic App version "
+                                          "for this host.\n")
+                elif "Is your docker daemon up and running" in e.output or \
+                     "Are you trying to connect to a TLS-enabled daemon without TLS" \
+                     in e.output:
+                    raise DockerException("Could not connect to the docker daemon.")
```

Comment:
@cdrage +1, it makes the code a lot cleaner and also more verbose to the end user.
Do you mean something like:
```python
        if not dryrun:
            try:
                subprocess.check_output([docker_cli, 'version'],
                                        stderr=subprocess.STDOUT)
            except subprocess.CalledProcessError as e:
                raise DockerException(e.output)
```
The output for this is,
```bash
...
...
2016-03-08 01:08:16,441 - [DEBUG] - nulecule/main.py - NuleculeManager init image: foo
2016-03-08 01:08:16,441 - [DEBUG] - nulecule/main.py - Request to unpack to foo to /var/lib/atomicapp/foo-d48b86bf8730
2016-03-08 01:08:16,441 - [INFO] - nulecule/base.py - Unpacking image: foo to /var/lib/atomicapp/foo-d48b86bf8730
2016-03-08 01:08:16,486 - [ERROR] - cli/main.py - Client:
 Version:      1.8.2-el7.centos
 API version:  1.20
 Package Version: docker-1.8.2-10.el7.centos.x86_64
 Go version:   go1.4.2
 Git commit:   a01dc02/1.8.2
 Built:        
 OS/Arch:      linux/amd64
Get http:///var/run/docker.sock/v1.20/version: dial unix /var/run/docker.sock: no such file or directory.
* Are you trying to connect to a TLS-enabled daemon without TLS?
* Is your docker daemon up and running?
```